### PR TITLE
Add Mergesort Benchmark

### PR DESCRIPTION
**Pull request by Mr4k**

### CODE REVIEW

**Comment by Mr4k** [comment]
Screen Shot 2020-11-03 at 12 58 02 PM
Not much better but it shows some improvement. I think we can bring this down as we make the language and caching logic better. I think right now it's a cool starting point to beat.

File: benchmark/benchmark_test.go [hunk 1]
```diff
@@ -50,3 +50,53 @@ func BenchmarkPureCollatz(b *testing.B) {
...
+	   }
+   }
+   repeat_merge_sort_with_modifications(200, ra)
+	`)
+	for i := 0; i < b.N; i++ {
+		evaluator.Eval(program, env)
+	}
+}
+
+func BenchmarkPureMergeSort(b *testing.B) {
```
Comment (Mr4k) [inline benchmark/benchmark_test.go:79]: Only the merge_sort function here is pure. The other functions were actually slowing it down (I'm guess by caching on results that were rarely used).

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 bench:
-	cd benchmark && go test -bench=. -benchtime=10s
```
Comment (Mr4k) [inline Makefile:2]: upped this b/c merge sort benchmark is longer

File: benchmark/benchmark_test.go (hunk 1 above)
Comment (jemmaissroff) [reply benchmark/benchmark_test.go:79]: This is neat because we can use it to improve our caching logic!

**Review by jemmaissroff — APPROVED** [review]

**Pull request merged**